src/Data/HashTable/ST/Basic.hs

```diff
@@ -620,6 +681,12 @@ recordIsDeleted = (== deletedMarker)
 
 
 ------------------------------------------------------------------------------
+{-# INLINE recordIsFilled #-}
+recordIsFilled :: Elem -> Bool
+recordIsFilled el = not (recordIsEmpty el) && not (recordIsDeleted el)
```

Comment:
That looks good. Are we always branching on the result of this function? If so, the "Bool" return value is ok, otherwise I'm wondering if we shouldn't use `Int#` instead if it would allow us to remove branches elsewhere.

> I'm just trying to minimize the number of commits I make before I get this patch right. :)

Appreciated, but they're not a scarce resource :)